src/main/mw.c

```diff
@@ -681,6 +685,112 @@ void processRx(void)
 
 }
 
+void getArmingChannel(modeActivationCondition_t *modeActivationConditions, uint8_t *armingChannel) {
+    for (int index = 0; index < MAX_MODE_ACTIVATION_CONDITION_COUNT; index++) {
+        modeActivationCondition_t *modeActivationCondition = &modeActivationConditions[index];
+        if (modeActivationCondition->modeId == BOXARM && IS_RANGE_USABLE(&modeActivationCondition->range)) {
+            *armingChannel = modeActivationCondition->auxChannelIndex +  NON_AUX_CHANNEL_COUNT;
+            break;
+        }
+    }
+}
+
+void filterRc(void){
+    static int16_t lastCommand[4] = { 0, 0, 0, 0 };
+    static int16_t deltaRC[4] = { 0, 0, 0, 0 };
+    static int16_t loop[5] = { 0, 0, 0, 0, 0 };
+    static int16_t factor, rcInterpolationFactor, loopAvg;
+    static uint32_t rxRefreshRate;
+    static int16_t lastAux, deltaAux;                            // last arming AUX position and delta for arming AUX
+	static uint8_t auxChannelToFilter;                           // AUX channel used for arming needs filtering when used
+	static int loopCount;
+
+    // Set RC refresh rate for sampling and channels to filter
+    if (!rxRefreshRate) {
+        if (feature(FEATURE_RX_PARALLEL_PWM | FEATURE_RX_PPM)) {
+            rxRefreshRate = 20000;
+
+            // AUX Channels to filter to replace PPM/PWM averaging
+            getArmingChannel(currentProfile->modeActivationConditions,&auxChannelToFilter);
+
+        }
+
+        // TODO Are there more different refresh rates?
+        else {
+    	    switch (masterConfig.rxConfig.serialrx_provider) {
+    	        case SERIALRX_SPEKTRUM1024:
+    	            rxRefreshRate = 22000;
+    	            break;
+    	        case SERIALRX_SPEKTRUM2048:
+    	            rxRefreshRate = 11000;
+    	            break;
+    	        case SERIALRX_SBUS:
+    	            rxRefreshRate = 11000;
+    	            break;
+    	        default:
+    	            rxRefreshRate = 10000;
+    	            break;
+            }
+        }
+
+        rcInterpolationFactor = 1; // Initial Factor before looptime average is calculated
+
+    }
+
+    // Averaging of cycleTime for more precise sampling
+    loop[loopCount] = cycleTime;
+    loopCount++;
+
+    // Start recalculating new rcInterpolationFactor every 5 loop iterations
+    if (loopCount > 4) {
+        uint16_t tmp = (loop[0] + loop[1] + loop[2] + loop[3] + loop[4]) / 5;
+
+        // Jitter tolerance to prevent rcInterpolationFactor jump too much
+        if (tmp > (loopAvg + LOOP_DEADBAND) || tmp < (loopAvg - LOOP_DEADBAND))  {
```

Comment:
Maybe consider removing deadband - if noise is within deadband, used value will 'stick' to same random value inside deadband interval and stay there. This may cause changes in flight performance, different 'stick' values may be used for considerable time each ...
Without deadband, the jitter should average itself out (or simply use IIR lowpass filter and set time constant high enough) ..